extensions/in-memory/assetindex-memory/src/main/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryAssetIndex.java

```diff
@@ -152,11 +152,27 @@ public void accept(Asset asset, DataAddress dataAddress) {
         accept(new AssetEntry(asset, dataAddress));
     }
 
+    @Override
+    public Asset deleteById(String assetId) {
```

Comment:
As per the upstream issue 801 it is mentioned that it should throw `EdcPersistenceException ` if anything goes wrong (e.g. a CosmosException happens, network error, unreachable service,...) but I think right now it throws `EdcException` in case of `CosmosException`.

We can confirm this behavior with Paul if needed or keep it later for the upstream review however you like.

Reply:
Good catch! I prefer to fix it in this PR now. Thanks.